### PR TITLE
Add Duktape JavaScript Interpreter

### PR DESCRIPTION
**Pull request by TheBrokenRail**
This might be useful for devices with less storage space.

### CODE REVIEW

**Comment by TheBrokenRail** [comment]
@xeffyr

File: packages/duktape/build.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+termux_step_make() {
+	make libduktape.so.1.0.0 duk CC=${CC} GXX=${CXX}
+}
+
+termux_step_make_install() {
+	cp libduktape.so.1.0.0 ${TERMUX_PREFIX}/lib/libduktape.so
+	cp duk ${TERMUX_PREFIX}/bin
+}
```
Comment (xeffyr) [inline packages/duktape/build.sh]: You are installing only library and program. Headers are not needed here ?
Something like
```
include/duk_config.h
include/duktape.h
```

File: packages/duktape/build.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+TERMUX_PKG_BUILD_IN_SRC=yes
+
+termux_step_extract_package() {
+	git clone --depth=1 https://github.com/svaarala/duktape.git -b v${TERMUX_PKG_VERSION} ${TERMUX_PKG_SRCDIR}
+}
+
+termux_step_post_extract_package() {
+	sudo apt-get -yq update
+	sudo apt-get install -yq python2.7 python-pip
+	pip install pyyaml
```
Comment (xeffyr) [inline packages/duktape/build.sh]: Privileged (sudo) package installation is not allowed for build scripts.

Comment (xeffyr) [reply packages/duktape/build.sh]: Packages `python2.7` and `python-pip` should be installed in Docker image.
See script [scripts/setup-ubuntu.sh](https://github.com/termux/termux-packages/blob/master/scripts/setup-ubuntu.sh).

Comment (TheBrokenRail) [reply packages/duktape/build.sh]: Done!

File: packages/duktape/build.sh (hunk 1 above)
Comment (TheBrokenRail) [reply packages/duktape/build.sh]: Done!

**Comment by TheBrokenRail** [comment]
Fixed!

**Pull request merged**